File: leap/leap.go
// Package leap tests for leap years
package leap

// IsLeapYear will return true if the given year is a leap year, else false
func IsLeapYear(year int) bool {

	if year%4 != 0 {
		return false
	}

	if year%100 == 0 {
		if year%400 == 0 {
			return true
		}
		return false
	}

	return true

}

File: solutions/csharp/leap/1/Leap.cs
public static class Leap
{
    public static bool IsLeapYear(int year) => year switch
    {
        _ when year % 400 == 0 => true,
        _ when year % 100 == 0 => false,
        _ when year % 4   == 0 => true,
        _                      => false,
    };
}